build/commands/lib/gitPatcher.js

```diff
@@ -177,6 +193,17 @@ module.exports = class GitPatcher {
     const prepOps = []
     this.logProgress(os.EOL + 'Getting patch data...')
     for (const patchData of patchesToApply) {
+      if (patchData.reason === patchApplyReasons.SRC_REMOVED) {
+        // Skip patches that the target file is gone, and flag them as
+        // failures. This is necessary to filter out these patches, so they are
+        // not passed along to `git reset`, as that causes an early bailing out
+        // by git, failing to reset any files listed after.
+        prepOps.push({
+          error: new Error('Target file does not exist'),
+          ...patchData
+        })
```

Comment:
preOps is an array of Promises, so you should push a resolved Promise instead: `preOps.push(Promise.resolve({ error: ... }))`.
But in this case I'm not sure we always want to report an error, because users who update to a version where both the brave .patch and the chromium target file are removed could see errors? When really it's just a cleanup and the .patchinfo is removed (in `handleObsoletePatchInfos`). 
Perhaps only report an error when the .patch file is still present, as I assume that's the case this PR is attempting to solve.

Reply:
> Perhaps only report an error when the .patch file is still present

So you mean I should also check for the existence of the patch file? I see.

Reply:
I believe it works fine at the moment when both the chromium destination file and the brave patch file are removed - we must have had that happen quite a bit.

And I understand you want to fix it (and get a nice error report) when a chromium file has been deleted but we still have a patch file.

Reply:
Added a test as well.